http/logging_test.go

```diff
@@ -23,20 +25,32 @@ var _ = Describe("Logging", func() {
 		defer ts.Close()
 		_, err = http.Get(ts.URL + "/logging")
 		Expect(err).NotTo(HaveOccurred())
-		Expect(strings.Contains(buf.String(), "completed handling request")).To(BeTrue())
+		Expect(strings.Contains(buf.String(), "successfully handled request")).To(BeTrue())
 	})
 
 	It("should support websockets", func() {
-		buf := &bytes.Buffer{}
+		buf := &Buffer{}
 		logrus.SetOutput(buf)
 		srv, err := createServer([]Option{WithLogging("test")})
 		Expect(err).NotTo(HaveOccurred())
 		ts := httptest.NewServer(srv.Handler)
 		defer ts.Close()
-
 		err = testWebsocketEcho(ts.URL)
 		Expect(err).NotTo(HaveOccurred())
-
-		Expect(strings.Contains(buf.String(), "completed handling request")).To(BeTrue())
+		time.Sleep(100 * time.Millisecond)
```

Comment:
why does it need a sleep? shouldn't the code above it be synchronous?

Reply:
No its unfortunately not. The `Hijack()` hands over the requests connection to another goroutine and the http server imediately returns its handler but explicitly does not close the connection.
Thats why the whole process gets async and we need to wait a bit for the logs to appear in our buffer.